index.js

```diff
@@ -319,19 +321,15 @@ module.exports = {
     }
 
     function listMembers(groupId, opts = {}) {
+      const { live } = opts
+
       return pull(
-        getPickedEpoch(groupId, { live: !!opts?.live }),
-        pull.map((pickedEpoch) => {
-          console.log('pickedEPoch', pickedEpoch)
-          return getMemberUpdates(pickedEpoch.id, { live: !!opts?.live })
-        }),
+        getPreferredEpoch.stream(groupId, { live }),
```

Comment:
i guess my worry about not having `getPickedEpoch` is that, since we use the picked key when posting, but then use `getPreferredEpoch` for everything else, there can in theory be race conditions where a new epoch is preferred but we haven't had time to pick it.

that's why i was thinking `getPreferredEpoch` fitted better as a thing in the background. it's fine if it's slow/hasn't caught up to everything yet, since everything related to posting/writing is all based on the picked key.

Reply:
yes the question is "what is the source of truth" ?

I hadn't fully understood what you were going for with picked vs preferred. perhaps we could check if the preferredEpoch is in the keystore before using it.

Reply:
I think I agree with you. I think I disagree that it's something for `lib/epochs` because you're not calculating anything with epochs, you're just using the latest write keys? But maybe I don't have the full overview for how you were using this.

Keen to discuss tonight

Reply:
> perhaps we could check if the preferredEpoch is in the keystore before using it.

we have `publish()` where we could add a check but we also use db.create in tribes2.create... although that is once, and at that point which key to use is really obvious

Reply:
added todo at the top